### PR TITLE
Feature/document commands

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

**Comment by bainternet** [comment]
@iNewLegend 

Seem to be missing propper depreciation of `elementorCommon.helpers` methods.

**Comment by iNewLegend** [comment]
> elementorCommon.helpers
findViewRecursive
findViewById

is new and never used before, please look here:
https://github.com/elementor/elementor/commit/ea9e26cdaaf8008754000524e3e8065c3d37577a#diff-6da925c663ffbe86af4aaf90394853ee

File: assets/dev/js/editor/document/history/commands/add-transaction.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import Base from '../../commands/base/base';
+
+export class AddTransaction extends Base {
```
Comment (matipojo) [inline assets/dev/js/editor/document/history/commands/add-transaction.js]: Rename. remove the term "transaction" from all places.

File: assets/dev/js/editor/document/history/commands/add-transaction.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	apply( args ) {
+		const currentId = elementor.history.history.getCurrentId();
```
Comment (matipojo) [inline assets/dev/js/editor/document/history/commands/add-transaction.js]: Why it should use historyId?

File: assets/dev/js/editor/document/history/commands/end-log.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import Base from '../../commands/base/base';
+
+export class EndLog extends Base {
+	apply( args ) {
+		if ( args.id ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/history/commands/end-log.js]: `id` is optional.

File: assets/dev/js/editor/document/history/commands/end-transaction.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		}
+
+		const firstItem = this.component.transactions[ 0 ],
+			{ type } = firstItem,
+			transactions = this.component.mergeTransactions( this.component.transactions );
+
+		let { title = '', subTitle = '' } = firstItem;
+
+		if ( transactions.length > 1 ) {
+			title = 'Elements'; // TODO: translate.
```
Comment (matipojo) [inline assets/dev/js/editor/document/history/commands/end-transaction.js]: Translate.

File: assets/dev/js/editor/elements/views/base.js [hunk 5]
```diff
@@ -201,10 +201,6 @@ BaseElementView = BaseContainer.extend( {
 	initialize() {
 		BaseContainer.prototype.initialize.apply( this, arguments );
 
-		if ( this.collection ) {
-			this.listenTo( this.collection, 'add remove reset', this.onCollectionChanged, this );
```
Comment (matipojo) [inline assets/dev/js/editor/elements/views/base.js]: Check other `onCollectionChanged`.

File: core/common/assets/js/components/events-hooks.js [hunk 6]
```diff
@@ -45,14 +46,27 @@ export default class EventsHooks extends elementorModules.Module {
...
+
 	checkEvent( event ) {
 		if ( -1 === Object.keys( this.callbacks ).indexOf( event ) ) {
 			throw Error( `${ this.getType() }: '${ event }' is not available.` );
 		}
 	}
 
 	checkId( id ) {
-		if ( 0 === this.usedIds.indexOf( id ) ) {
+		if ( this.usedIds.indexOf( id ) > 0 ) {
```
Comment (matipojo) [inline core/common/assets/js/components/events-hooks.js]: `!== -1` ?

File: core/common/assets/js/utils/helpers.js [hunk 7]
```diff
@@ -37,41 +37,4 @@ export default class Helpers {
 			return $1.toUpperCase();
 		} );
 	}
-
-	findViewRecursive( parent, key, value, multiple = true ) {
```
Comment (matipojo) [inline core/common/assets/js/utils/helpers.js]: add `elementorCommon.helpers.softDeprecated` message.

File: core/common/assets/js/utils/helpers.js [hunk 8]
```diff
@@ -37,41 +37,4 @@ export default class Helpers {
...
-					}
-				}
-			}
-		}
-
-		return found;
-	}
-
-	findViewById( id ) {
-		const elements = this.findViewRecursive(
```
Comment (matipojo) [inline core/common/assets/js/utils/helpers.js]: add `elementorCommon.helpers.softDeprecated` message.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/utils/helpers.js (hunk 8 above)
Comment (iNewLegend) [reply core/common/assets/js/utils/helpers.js]: is new and never used before.

File: core/common/assets/js/utils/helpers.js (hunk 7 above)
Comment (iNewLegend) [reply core/common/assets/js/utils/helpers.js]: is new and never used before.

**Review by matipojo — DISMISSED** [review]

**Pull request merged**